### PR TITLE
Return ConfirmedBlock value from Storage

### PR DESCRIPTION
**Pull request by deuszx**
## Motivation

Even though the only type of `CertificateValue` we ever store in a persistent storage is a `ConfirmedBlock` variant - the API of `Storage` trait worked with `HashedCertificateValue` type which has three variants - `ValidatedBlock`, `Timeout` and `ConfirmedBlock`.

## Proposal

Change the API of `Storage` trait to return a `*ConfirmedBlock` type.

## Test Plan

CI should catch any regressions.

## Release Plan

- Nothing to do / These changes follow the usual release cycle.

## Links

Closes https://github.com/linera-io/linera-protocol/issues/2968

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-chain/src/block.rs [hunk 1]
```diff
@@ -63,6 +68,42 @@ pub struct ConfirmedBlock {
...
+    }
+
+    #[graphql(derived(name = "value"))]
+    async fn _value(&self) -> ConfirmedBlock {
+        self.inner().clone()
+    }
+}
+
+impl TryFrom<CertificateValue> for ConfirmedBlock {
+    type Error = &'static str;
```
Comment (afck) [inline linera-chain/src/block.rs:97]: Should this be a `ConversionError` now, as in https://github.com/linera-io/linera-protocol/pull/2951?

**Review by afck — APPROVED** [review]

File: linera-chain/src/block.rs (hunk 1 above)
Comment (deuszx) [reply linera-chain/src/block.rs:97]: This PR is part of the bigger stack and I haven't merged `main` in yet. Will do.

File: linera-storage/src/db_storage.rs [hunk 2]
```diff
@@ -404,12 +401,9 @@ where
     async fn read_hashed_certificate_value(
```
[the remark this answers is not in the document]
Comment (deuszx) [reply linera-storage/src/db_storage.rs]: Done in [s/read_hashed_certificate_value/read_hashed_confirmed_block/g](https://github.com/linera-io/linera-protocol/pull/2950/commits/f3daeac26045118479ca97ad3e80f71e5f71ad6e)

File: linera-chain/src/block.rs (hunk 1 above)
Comment (deuszx) [reply linera-chain/src/block.rs:98]: Done in [Merge remote-tracking branch 'origin/main' into storage-read-returns-…](https://github.com/linera-io/linera-protocol/pull/2950/commits/a71918bf501e9ad9126919d182f3eecc5405216e) (although it seems to bring in two commits oO).

**Review by afck — APPROVED** [review]

File: linera-storage/src/db_storage.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (afck) [reply linera-storage/src/db_storage.rs]: (Thanks! I'd have omitted the `hashed`, but I'm fine with this, too.)

**Pull request merged**